sound/soc/soc-topology.c

```diff
@@ -1156,18 +1156,22 @@ static int soc_tplg_dapm_graph_elems_load(struct soc_tplg *tplg,
 			SNDRV_CTL_ELEM_ID_NAME_MAXLEN)
 			return -EINVAL;
 
-		route.source = elem->source;
-		route.sink = elem->sink;
-		route.connected = NULL; /* set to NULL atm for tplg users */
+		route = kzalloc(sizeof(*route), GFP_KERNEL);
+		if (!route)
+			return -ENOMEM;
```

Comment:
Where do we unwind this if it fails. i.e say count is 10 and route 8 fails kzalloc. Where do you kfree routes 0 - 7 ?

Reply:
I think this is a good question. 
each time when the route is allocated successfully, it will be linked into the list.
in the topology free stage, the route elem will be got from the list and do the free.
from your question, I think maybe this PR is not enough.
so from your question, I propose a concern of design:
if one of the routes allocated failure, should we free all of them and block this topology parsing or just skip this one and continue for next?

Reply:
we want to stop parsing on any error (whether it comes from the topology file or memory allocation). There's no point in trying to conceal and compensate for errors, it's going to lead to more issues.

Reply:
you should only free the routes allocated by this invocation of this function. In this case routes 0 - 7 and the strdup names for routes 0 - 7 too.

Reply:
ok. no problem